black.py

```diff
@@ -1364,27 +1346,83 @@ def preceding_leaf(node: Optional[LN]) -> Optional[Leaf]:
     return None
 
 
-def is_delimiter(leaf: Leaf) -> int:
-    """Return the priority of the `leaf` delimiter. Return 0 if not delimiter.
+def is_split_after_delimiter(leaf: Leaf, previous: Leaf = None) -> int:
+    """Returns the priority of the `leaf` delimiter, given that it is a delimiter
+    that would cause a line break after the delimiter. Returns 0 if not a
+    delimiter in this case.
 
     Higher numbers are higher priority.
     """
     if leaf.type == token.COMMA:
         return COMMA_PRIORITY
 
-    if leaf.type in COMPARATORS:
-        return COMPARATOR_PRIORITY
+    if (
+        leaf.type in KWARGS
+        and leaf.parent
+        and leaf.parent.type in {syms.argument, syms.typedargslist}
+    ):
+        return KWARGS_PRIORITY
```

Comment:
I actually don't quite get why you added this. Can you add a unit test that showcases what splitting on stars does?